Fhi.HelseId.Web/Services/HelseIdSecretHandlers.cs

```diff
@@ -23,14 +23,33 @@ public interface IHelseIdSecretHandler
 
     public abstract class SecretHandlerBase : IHelseIdSecretHandler
     {
-        protected JsonWebKey? jwkSecurityKey;
-        protected IHelseIdWebKonfigurasjon? configAuth;
+        protected JsonWebKey _jwkSecurityKey = null!; // Explicitly checks for null when GenerateClientAssertion is called.
+        protected IHelseIdWebKonfigurasjon _configAuth = null!; // Explicitly checks for null when GenerateClientAssertion is called.
```

Comment:
These looks like the scream "design issue".  I know they are old, but the fact we create the secrethandler without what it needs to work, which is filled at a later stage opens this can of worms.  So we get protected fields, that we need to state will never be null, which actually is not true.  They depend on a public method to be called, and that method has to set these two, which may not happen since the method in question is virtual.  

That also calls for the null checks below which doesnt look good.

Suggest we rewrite the secrethandler design and fix the calling extensions methods which uses these.